pkg/k8s/watchers/watcher.go

```diff
@@ -447,6 +443,14 @@ func (k *K8sWatcher) resourceGroups() (beforeNodeInitGroups, afterNodeInitGroups
 		k8sAPIGroupNodeV1Core,
 	}
 
+	if k.cfg.K8sNetworkPolicyEnabled() {
+		// When the flag is set,
+		// We need all network policies in place before restoring to
+		// make sure we are enforcing the correct policies for each
+		// endpoint before restarting.
+		k8sGroups = append(k8sGroups, k8sAPIGroupNetworkingV1Core)
+	}
```

Comment:
Should we enforce that `enable-policy=never` be set whenever this flag is passed? I'm a bit worried that someone might enable this and disable policy enforcement by mistake.

Reply:
this seems like a good idea to me as well

Reply:
@pchaigno 
I mark this new flag as hidden as we discussed in the slack.